compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -758,67 +881,101 @@ static LogicalResult generateUnifiedObject(
   std::string errorMessage;
   {
     auto output = openOutputFile(LLVMIRFile, &errorMessage);
-    if (!output) return moduleOp.emitOpError(errorMessage);
+    if (!output) {
+      llvm::errs() << "failed to open output LLVMIRFile because: "
+                   << errorMessage;
+      return failure();
+    }
     llvmModule->print(output->os(), nullptr);
     output->keep();
   }
 
   if (useChess) {
-    SmallString<64> chessWrapperBin(mlirAIEInstallDir);
-    sys::path::append(chessWrapperBin, "bin", "xchesscc_wrapper");
-
-    SmallString<64> chessworkDir(tempDir);
-    sys::path::append(chessworkDir, "chesswork");
-
-    SmallString<64> chessIntrinsicsLL(mlirAIEInstallDir);
-    sys::path::append(chessIntrinsicsLL, "aie_runtime_lib",
-                      StringRef(targetArch).upper(),
-                      "chess_intrinsic_wrapper.ll");
+    auto aieToolsDir = findVitis(vitisDir);
+    if (!aieToolsDir) {
+      llvm::errs() << "Failed to find vitis";
+      return failure();
+    }
 
-    std::string llvmirString;
+    SmallString<64> inputLLChessHackedFile(tempDir);
+    sys::path::append(inputLLChessHackedFile, "input.chesshacked.ll");
     {
-      raw_string_ostream llvmirStream(llvmirString);
-      llvmModule->print(llvmirStream, nullptr);
+      std::string inputLLStr;
+      auto inputLL = openInputFile(LLVMIRFile, &errorMessage);
+      if (!inputLL) {
+        llvm::errs() << "Failed to open input.ll file because: "
+                     << errorMessage;
+        return failure();
+      }
+      inputLLStr = std::string(inputLL->getBuffer());
+      std::string inputLLChessHackedStr = chesshack(inputLLStr);
```

Comment:
This still uses the existing `chesshack` approach, right? Not the LLVM 15/17? dialect approach?

Reply:
Correct - I didn't do that here (and haven't done it) because I haven't quite decided where those libs should come from (LLVM submodule checked out to the right commit? `ExternalProject_Add` with a zip? Pre-built and kept somewhere?).